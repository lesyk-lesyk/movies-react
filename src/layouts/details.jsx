import * as React from "react";
import { withRouter } from 'react-router';
import PropTypes from 'prop-types'

import { Header } from '../components/header';
import { Content } from '../components/content';
import { Footer } from '../components/footer';

import { Button } from '../components/button';

import { MovieDetails } from '../components/movie-details';
import { SubHeader } from '../components/sub-header';
import { MovieList } from '../components/movie-list'
import './details.scss';

export class Details extends React.Component {
  static propTypes = {
    match: PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired
  }

  constructor() {
    super();
    this.returnToSearh = this.returnToSearh.bind(this);
    this.movies = [];
    this.movie = {};
  }

  returnToSearh() {
    this.props.history.push('/');
  }

  render() {
    console.log(this.props.match.params.id);
    return (
      <React.Fragment>

        <Header>
          <Button
            size="sm"
            color="white"
            className="top-search-button"
            onClick={this.returnToSearh}>
            Search
          </Button>
          <MovieDetails movie={this.movie} />
        </Header>
        <SubHeader>
          Films by {this.genre} genre
        </SubHeader>

        <Content>
          <MovieList movies={this.movies} />
        </Content>

        <Footer />

      </React.Fragment>
    );
  }
}

export const DetailsLayout = withRouter(Details);
