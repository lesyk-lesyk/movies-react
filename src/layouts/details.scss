.top-search-button{
  float: right;
  position: relative;
  top: 20px;
}
