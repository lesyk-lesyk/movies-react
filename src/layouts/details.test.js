import React from 'react';
import { DetailsLayout } from './details';
import renderer from 'react-test-renderer';

jest.mock('./details.scss', () => ({}));
jest.mock('../components/header/header');
jest.mock('../components/content/content');
jest.mock('../components/footer/footer');
jest.mock('../components/movie-details/movie-details');
jest.mock('../components/sub-header/sub-header');
jest.mock('../components/movie-list/movie-list');

describe('DetailsLayout', () => {
  it('renders', () => {
    const tree = renderer
      .create(<DetailsLayout />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
