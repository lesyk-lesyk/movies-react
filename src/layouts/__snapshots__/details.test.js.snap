// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsLayout renders 1`] = `
Array [
  <Header>
    <MovieDetails />
    <SubHeader>
      Films by 
      Dramma
       genre
    </SubHeader>
  </Header>,
  <Content>
    <MovieList />
  </Content>,
  <Footer />,
]
`;
